health_monitor/src/health_monitor.cpp

```diff
@@ -139,8 +150,138 @@ namespace health_monitor
         ros::Duration sd(startup_duration_);
         long start_duration=sd.toNSec() / 1e6;
 
-        nh_.publishSystemAlert(driver_manager_.handleSpin(truck_,car_,time_now,start_up_timestamp_,start_duration,start_time_flag_.isZero()));
+        
+        /*Publish each system alert only once*/
+        auto crit_driver_status_truck = driver_manager_.are_critical_drivers_operational_truck(time_now);
+        auto crit_driver_status_car = driver_manager_.are_critical_drivers_operational_car(time_now);
+
+
+        auto dm = driver_manager_.handleSpin(truck_,car_,time_now,start_up_timestamp_,start_duration,start_time_flag_.isZero());
+        if(truck_ == true)
+        {
+            if((crit_driver_status_truck == "s_1_l1_1_l2_1_g_1") && (is_published_truck[0] == false)) //"All essential drivers are ready"
+            {
+             	nh_.publishSystemAlert(dm);
+
+  		        is_published_truck[0] = true;
+            } 
+           else if((start_time_flag_.isZero()==true) || (time_now - start_up_timestamp_ <= start_duration) && (is_published_truck[1] == false)) //"System is starting up..."
+            {
+                nh_.publishSystemAlert(dm);
+
+  		        is_published_truck[1] = true;
+            } 
+            else if((crit_driver_status_truck == "s_1_l1_0_l2_1_g_1") || (crit_driver_status_truck=="s_1_l1_1_l2_0_g_1")&& (is_published_truck[2] =false)) //One LIDAR Failure
```

Comment:
In this if stmt and next, the third test is an assignment, not a comparison operator!  It will always return true.  Using a single = instead of a double == in if tests is a common C++ mistake (easy typo).  Therefore, it is advantageous to use the power of booleans - they don't need to be compared to anything!  if (a) { ... }   or   if (!a) { ... }.  This not only avoids the possibility of mistyping a ==, it also makes the code a lot more readable.